frontend/src/components/general/form/types/FileUpload.tsx

```diff
@@ -47,21 +51,33 @@ const FileUpload: React.FC<FileUploadProps> = ({
 
         <button
           type="button"
-          className={`${showDeleteButton ? 'block' : 'hidden'} `}
+          className={`${isFileSelected ? 'block' : 'hidden'} `}
           onClick={handleClearInput}
         >
           <Trash2 className="text-dark-600" />
           <span className="sr-only">Datei unselektieren</span>
         </button>
       </div>
 
-      {message && (
-        <div className="mt-4 text-red font-semibold text-sm">
+      {message && !isFileSelected && (
+        <div className={`mt-4 ${getClassNameForMessageType(messageType)}
+        font-semibold text-sm`}>
           {message}
         </div>
       )}
     </>
   );
 };
-
-export default FileUpload;
+const getClassNameForMessageType = (messageType: MessageType) => {
```

Comment:
You do not need a switch case for this. You could directly set the color in the MessageType you created. You can take a look at the file `useDetailsForWateringStatus.ts`. In that file the same logic is used.